### PR TITLE
feature: full feature app

### PR DESCRIPTION
**Pull request by andela-obamidele**
check the app [here](http://aproko-international.herokuapp.com)
#### What does this PR do?
Contains a deployable version o the [app](http://aproko-international.herokuapp.com)
#### Description of Task to be completed?
code review and implementation of feedback
#### How should this be manually tested?
just run ```npm test```
#### Any background context you want to provide?
read the documentation for more infomation
#### What are the relevant pivotal tracker stories?
None
#### Screenshots (if appropriate)
None
#### Questions:
None

### CODE REVIEW

File: src/components/Sources.jsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    Actions.getSources();
+    sourcesStore.on('change', this.getAllNewsSources);
+  }
+  /**
+   * @description removes all listeners before unmounting
+   * @returns {void}
+   */
+  componentWillUnmount() {
+    sourcesStore.removeListener('change', this.getAllNewsSources);
+    const searchBox = this.refs.searchBox;
```
Comment (andela-oaladeusi) [inline src/components/Sources.jsx]: Great work here
It's a good programming practice to declare variables at the beginning of every function for readability purpose.

File: src/components/Sources.jsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    this.setState({ sources, loading: false });
+  }
+  /**
+   * @description Maps array of source object into array of source components
+   * @returns {JSX.Element[]} Array of news sources elements
+   */
+  makeActualSourcesComponents() {
+    let sources = this.state.sources;
+    sources = sources.map((source) => {
+      const sourceComponent = (<Source source={source} key={source.id} />);
```
Comment (andela-oaladeusi) [inline src/components/Sources.jsx]: This can be returned directly without declaring a variable for it

File: src/components/partials/Article.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+   * @return {JSX.Element} -div element containing article summary
+   */
+  render() {
+    const {
+      title,
+      description,
+      url,
+      urlToImage,
+      publishedAt
+    } = this.props.article;
```
Comment (andela-oaladeusi) [inline src/components/partials/Article.jsx]: Good work here

File: src/components/partials/ErrorDialog.jsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  componentWillUnmount() {
+    errorStore.removeEventListener('change', this.checkForError);
+  }
+  /**
+   * @description fetches data from component store
+   * @returns {void}
+   */
+  checkForError() {
+    const message = errorStore.getErrorMessage();
+    const shown = !!message;
```
Comment (andela-oaladeusi) [inline src/components/partials/ErrorDialog.jsx]: How does `!!message` works?

File: src/components/partials/NavigationBar.jsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import * as ReactBootstrap from 'react-bootstrap';
```
Comment (andela-oaladeusi) [inline src/components/partials/NavigationBar.jsx]: Hmmm...Why are you importing all `react-bootstrap` components here?

File: test/components/__snapshots__/Article.test.js.snap [hunk 6]
```diff
@@ -0,0 +1,68 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (andela-oaladeusi) [inline test/components/__snapshots__/Article.test.js.snap]: Not sure if you need to push `snapshot` files.

File: .codeclimate.yml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+engines: 
+  eslint:
+    enabled: true
+    
+ratings:
+   paths:
+   - "**.js"
+   - "**.jsx"
```
Comment (andela-obamidele) [inline .codeclimate.yml]: fixed

File: .coveralls.yml [hunk 8]
```diff
@@ -0,0 +1,2 @@
+repo_token: T2IpKXJfSTePDSuxNW0cJblNThA3n9bHj
+
```
Comment (andela-obamidele) [inline .coveralls.yml]: It is not a js file so there is no way I would get the token if  I put it in .env. but I figured out that the file should not even be pushed to github at all. so I'm just going to add it to my gitignore

File: .eslintrc [hunk 9]
```diff
@@ -1,100 +1,82 @@
...
+    }],
+    "require-jsdoc": [1, {
+      "require": {
+        "FunctionDeclaration": true,
+        "MethodDefinition": false,
+        "ClassDeclaration": false
+      }
+    }]
   }
 }
```
Comment (andela-obamidele) [inline .eslintrc]: fixed

File: src/actions/Actions.js [hunk 10]
```diff
@@ -0,0 +1,66 @@
+import dispatcher from '../dispatcher';
+import HeadlinesUtils from '../utils/headlines-utils';
+
+// Did this to improve readability
+const HU = HeadlinesUtils;
```
Comment (andela-obamidele) [inline src/actions/Actions.js]: fixed

File: src/components/partials/ErrorDialog.jsx (hunk 4 above)
Comment (andela-obamidele) [inline src/components/partials/ErrorDialog.jsx]: so it turns it into really boolean value rather.  so lets say ```message = 'this is a message```
```!!message``` becomes true.

File: src/components/partials/NavigationBar.jsx (hunk 5 above)
Comment (andela-obamidele) [inline src/components/partials/NavigationBar.jsx]: obviously, I have grown past this. I'll make changes now

File: test/components/__snapshots__/Article.test.js.snap (hunk 6 above)
Comment (andela-obamidele) [inline test/components/__snapshots__/Article.test.js.snap]: I've ignored it already. why it is still showing

Comment (andela-oaladeusi) [inline test/components/__snapshots__/Article.test.js.snap]: I guess it's because you'd pushed it before adding it to `.gitignore`

**Pull request merged**